src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -1197,10 +1199,12 @@ void SystemDumpMapDCmd::execute(DCmdSource source, TRAPS) {
       output()->print_cr("(NMT is disabled, will not annotate mappings).");
     }
     MemMapPrinter::print_all_mappings(&fs);
+#ifndef _WIN64
     // For the readers convenience, resolve path name.
     char tmp[JVM_MAXPATHLEN];
     const char* absname = os::Posix::realpath(name, tmp, sizeof(tmp));
     name = absname != nullptr ? absname : name;
+#endif
```

Comment:
I wince a bit a the new ifdefs. Possibly for a separate RFE, it would be nice to have a function os::realpath() that uses realpath(2) on POSIX platforms, _fullpath on Windows.

Reply:
I agree, and was surprised this wasn't originally implemented.  I'll prepare a second PR.

Reply:
I have filed issue [#8338851](https://bugs.openjdk.org/browse/JDK-8338851) "Hoist os::Posix::realpath() to os::realpath() and implement on Windows" and will submit a PR shortly.

Reply:
I have created PR #20683 to address this.